src/plugins/validation/oas3/semantic-validators/security-definitions-ibm.js

```diff
@@ -0,0 +1,191 @@
+
+  for (const key in securitySchemes) {
+    const security = securitySchemes[key];
+    const type = security.type;
+    const path = `securitySchemes.${key}`;
+
+    if (!type) {
+      errors.push({
+        message: `'type' must be defined as a string and is required for path: ${path}`,
+        path,
+        authId: key
+      });
+    } else if (auths.indexOf(type) === -1) {
+      errors.push({
+        message:
+          '`type` must have one of the following types: `apiKey`, `oauth2`, `http`, `openIdConnect`',
+        path,
+        authId: key
+      });
+    } else {
+      //apiKey validation
+      if (type === API_KEY) {
+        const authIn = security.in;
+        if (!authIn) {
+          errors.push({
+            message:
+              "apiKey authorization must have required 'in' property, valid values are 'query' or 'header' or 'cookie'.",
+            path,
+            authId: key
+          });
+        } else if (
+          authIn !== 'query' &&
+          authIn !== 'header' &&
+          authIn != 'cookie'
+        ) {
+          errors.push({
+            message:
+              "apiKey authorization must have required 'in' property, valid values are 'query' or 'header' or 'cookie'.",
+            path,
+            authId: key
+          });
+        }
+      }
+
+      if (type === API_KEY && !security.name) {
+        errors.push({
+          message:
+            "apiKey authorization must have required 'name' string property. The name of the header or query parameter to be used.",
+          path,
+          authId: key
+        });
+      }
+      // oauth2 validation
+      else if (type === OAUTH2) {
+        const flows = security.flows;
+        const tokenUrl = security.tokenUrl;
+
+        if (!flows) {
+          errors.push({
+            message:
+              "oauth2 authorization must have required 'flows'. Valid values are 'implicit', 'password', 'clientCredentials' or 'authorizationCode'", /////recieved
+            path,
+            authId: key
+          });
+        } else if (
+          !flows.implicit &&
+          !flows.authorizationCode &&
+          !flows.authorizationCode &&
+          !flows.clientCredentials
+        ) {
+          errors.push({
+            message:
+              "oauth2 authorization must have required 'flows'. Valid values are 'implicit', 'password', 'clientCredentials' or 'authorizationCode'", /////recieved
+            path,
+            authId: key
+          });
+        } else if (flows && flows.implicit) {
+          const authorizationUrl = flows.implicit.authorizationUrl;
+          if (!authorizationUrl) {
+            errors.push({
+              message:
+                "oauth2 authorization implicit flow must have required 'authorizationUrl' parameter if type is `implicit` or `authorizationCode`.", /////////exepected
+              path,
+              authId: key
+            });
+          } else if (
```

Comment:
I think we want this to be an independent if check -- not in the else of the previous.  We want to catch and report all the problems in the document.